tuna/worker_interface.py

```diff
@@ -234,22 +253,31 @@ def refresh_query_objects(self, session, rows):
       except TypeError:
         session.refresh(obj_tuple)
 
-  def job_queue_push(self, job_rows):
+  def job_queue_push(self, job_rows: List[Tuple[SimpleDict, ...]]) -> None:
     """load job_queue with info for job ids"""
-    for job_tuple in job_rows:
+    for job_tuple in job_rows:  #type: ignore
       self.job_queue.put(job_tuple)
-      job = self.get_job_from_tuple(job_tuple)
-      self.logger.info("Put job %s %s %s", job.id, job.state, job.reason)
+      job = self.get_job_from_tuple(job_tuple)  #type: ignore
+      if job is not None:
+        self.logger.info("Put job %s %s %s", self.job.id, self.job.state,
+                         self.job.reason)
 
-  def job_queue_pop(self):
+  def job_queue_pop(self) -> None:
     """load job from top of job queue"""
     self.job = self.job_queue.get(True, 1)[0]
     self.logger.info("Got job %s %s %s", self.job.id, self.job.state,
                      self.job.reason)
 
   #pylint: disable=too-many-branches
-  def get_job(self, find_state, set_state, imply_end):
+  def get_job(self, find_state: str, set_state: str, imply_end: bool) -> bool:
     """Interface function to get new job for builder/evaluator"""
+    job_rows: List[Tuple[SimpleDict, ...]]
+    job_tables: List[Tuple[SimpleDict, ...]]
+    job_set_attr: List[str]
+    session: DbSession
+    ids: list
+    row: str
```

Comment:
its not a string, its of JobMixin type i believe. It is a row in the Job table which is an implementation of the JobMixin interface